src/little_dict.jl

```diff
@@ -75,6 +75,9 @@ LittleDict(itr::T) where T = LittleDict{kvtype(eltype(T))...}(itr)
 LittleDict(itr1::Pair, itr2::Pair) = LittleDict(first.([itr1, itr2]), last.([itr1,itr2]))
 LittleDict(itr1::Pair) = LittleDict([first(itr1)], [last(itr1)])
```

Comment:
This line can be deleted, no?

Reply:
Yes, probably. The problem arises in the test specific to these lines. The definition of the declaration wasn't clear to me, so I was waiting for your review.

Reply:
```
julia> d = LittleDict((1, 2), ("a", "b"))
LittleDict{Int64,String,Tuple{Int64,Int64},Tuple{String,String}} with 2 entries:
  1 => "a"
  2 => "b"
```
removing the lines have this effect on the definition.

Reply:
That looks correct to me.
The `LittleDict(keys::Tuple, vals::Tuple)` constructor should take priority over:
`LittleDict(kv1, kv2)`
as `keys::Tuple, vals::Tuple` is the pattern to avoid allocations.

Reply:
Cool. But the tests have 
```
@test d[1] === 2
@test d["a"] == "b"
```
which should be omitted, on allowing this definition

Reply:
Yes, i think those tests were just wrong.
But apparently past-me disagreed.
So it bears more thinking.

Reply:
These tests are fine and have the correct behavour:

https://github.com/JuliaCollections/OrderedCollections.jl/blob/bc0407699942c1b92585080d4519556dc0fa2023/test/test_little_dict.jl#L273-L279

None of them are `LittleDict(::Tuple, ::Tuple)`
they are either `LittleDict(::Pair, ::Pair)` and `LittleDict(::Tuple)` and `LittleDict(::Vector)`

Reply:
Well, travis failed on these particular testset, so I assumed something must be off here itself.
And moreover, 
```
julia> d = LittleDict((1, 2), ("a", "b"))
LittleDict{Int64,String,Tuple{Int64,Int64},Tuple{String,String}} with 2 entries:
  1 => "a"
  2 => "b"
```
and the tests can't be simultaneously true, assuming if I drop 
```
LittleDict(itr1::Pair, itr2::Pair) = LittleDict(first.([itr1, itr2]), last.([itr1,itr2]))
LittleDict(itr1::Pair) = LittleDict([first(itr1)], [last(itr1)])
```

Reply:
> and the tests can't be simultaneously true, assuming if I drop

Why not?
There is no test for that.

The tests are:
```
 d = LittleDict(((1, 2), ("a", "b")))   # LittleDict(::Tuple)`
 d2 = LittleDict([(1, 2), ("a", "b")])   # LittleDict(::Vector)
 d3 = LittleDict(1 => 2, "a" => "b")   # LittleDict(::Pair, ::Pair)
 d4 = LittleDict((1 => 2, "a" => "b"))   # LittleDict(::Tuple)
```


`LittleDict((1, 2), ("a", "b"))` is `LittleDict(::Tuple, ::Tuple)`

Reply:
the problem arises due to 
`d = LittleDict(((1, 2), ("a", "b")))`
It's a `Tuple{Tuple{Int64,Int64},Tuple{String,String}}`.
if you glance at my implementation, this ctor call triggers a call to my ctor in which `K` is `Tuple{Int64, Int64}` and `V` is `Tuple{String, String}`. I'll probably hack into my implementation to see if I can work around without this line getting triggered. Probably, that will be sufficient.

Reply:
@oxinabox I've done a bit of hackery to handle the case. And you were right, I was worried about the wrong test case the whole time :sweat_smile: